### PR TITLE
APIGOV-21060 Create watch client as a job

### PR DESCRIPTION
**Pull request by tsjohns9**

### CODE REVIEW

File: pkg/agent/agent.go [hunk 1]
```diff
@@ -213,12 +215,11 @@ func OnAgentResourceChange(agentResourceChangeHandler ConfigChangeHandler) {
 }
 
 func startAPIServiceCache() {
-	var checkStatus hc.CheckStatus
-
 	// register the update cache job
 	newDiscoveryCacheJob := newDiscoveryCache(false)
 	if !agent.cfg.IsUsingGRPC() {
-		checkStatus = agent.apicClient.Healthcheck
+		hc.RegisterHealthcheck(util.AmplifyCentral, "central", agent.apicClient.Healthcheck)
```
Comment (jcollins-axway) [inline pkg/agent/agent.go:221]: Maybe this could/should be converted into a job as well?

File: pkg/agent/stream/resourceclient.go [hunk 2]
```diff
@@ -65,3 +66,28 @@ func (c *resourceClient) get(selfLink string) (*apiv1.ResourceInstance, error) {
...
+// 	req.Headers["Content-Type"] = "application/json"
+// 	res, err := c.client.Send(req)
+// 	if err != nil {
+// 		return nil, err
+// 	}
+//
+// 	err = r.UnmarshalJSON(res.Body)
+//
+// 	return r, err
+// }
```
Comment (jcollins-axway) [inline pkg/agent/stream/resourceclient.go:93]: just dead for now?

File: pkg/apic/subscriptionmanager.go [hunk 3]
```diff
@@ -91,6 +91,10 @@ func (sm *subscriptionManager) Status() error {
 }
 
 func (sm *subscriptionManager) Execute() error {
+	if len(sm.statesToQuery) == 0 {
+		return nil
+	}
+
```
Comment (jcollins-axway) [inline pkg/apic/subscriptionmanager.go:97]: The 4 lines are no longer needed with the other updates in place

File: pkg/agent/stream/resourceclient.go (hunk 2 above)
Comment (tsjohns9) [reply pkg/agent/stream/resourceclient.go:93]: I'll come back to this. I started some work for a different branch on here too.

File: pkg/agent/agent.go (hunk 1 above)
Comment (tsjohns9) [reply pkg/agent/agent.go:223]: Maybe for another story

File: pkg/agent/stream/client.go [hunk 4]
```diff
@@ -1,87 +1,129 @@
...
-			status.Details = "the stream to central is not open"
-		}
-		log.Infof("Stream status: %s", status.Result)
-		return status
+// Status a health check endpoint for the connection to central.
+func (sc *Client) Status() error {
+	ok := sc.manager.Status()
+
+	if !ok {
+		return fmt.Errorf("grpc client is not connected to central")
```
[the remark this answers is not in the document]
Comment (tsjohns9) [reply pkg/agent/stream/client.go:89]: Where do we define those errors?

File: pkg/util/errors/errors.go [hunk 5]
```diff
@@ -7,4 +7,5 @@ var (
 	ErrPeriodicCheck             = Newf(1003, "%s failed.  Services are not ready")
 	ErrStartingAgentStatusUpdate = Newf(1004, "error starting %s update")
 	ErrStartingVersionChecker    = Newf(1005, "%s. No version to compare for upgrade")
+	ErrGrpcConnection            = Newf(1006, "grpc client is not connected to central")
```
Comment (jcollins-axway) [inline pkg/util/errors/errors.go:10]: this could just be New as there are no strung substitutions to do

**Pull request merged**